processing/src/main/java/io/druid/query/extraction/MapLookupExtractor.java

```diff
@@ -72,13 +73,19 @@ public String apply(@NotNull String val)
   @Override
   public List<String> unapply(final String value)
   {
-    return Lists.newArrayList(Maps.filterKeys(map, new Predicate<String>()
-    {
-      @Override public boolean apply(@Nullable String key)
+    if (this.map instanceof HashBiMap) {
+      return Lists.newArrayList(((HashBiMap<String, String>) this.map).inverse().get(value));
+    } else {
+      return Lists.newArrayList(Maps.filterKeys(map, new Predicate<String>()
```

Comment:
(minor) can this use java 8 lambdas like

`key -> map.get(key).equals(Strings.nullToEmpty(value))`

?

Reply:
Fixed